File: 김하영/the shortest path/11404.js
//백준 11404 플로이드

const fs = require("fs");

const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";

const input = fs.readFileSync(filePath).toString().trim().split("\n");

const N = +input.shift();
const M = +input.shift();

const graph = Array.from(Array(N), () => Array(N).fill(Infinity));

for (let i = 0; i < N; i++) {
  graph[i][i] = 0;
}

for (let i = 0; i < M; i++) {
  const [a, b, c] = input[i].split(" ").map((item) => +item);

  graph[a - 1][b - 1] = c;
}

const floyd = (graph) => {
  // 플로이드 와샬 알고리즘
  // k를 경유했을 때의 가중치
  // ex) k = 1, i = 4, j = 2일 때, 1을 경유해서 4에서 2로 가는 가중치
  // graph[4][2] = min(graph[4][2], graph[4][1] + graph[1][2])
  for (let k = 0; k < N; k++) {
    for (let i = 0; i < N; i++) {
      for (let j = 0; j < N; j++) {
        graph[i][j] = Math.min(graph[i][j], graph[i][k] + graph[k][j]);
      }
    }
  }
  return graph;
};

const result = floyd(graph);

for (let i = 0; i < N; i++) {
  for (let j = 0; j < N; j++) {
    if (result[i][j] === Infinity) {
      result[i][j] = 0;
    }
  }
}

for (let i = 0; i < N; i++) {
  console.log(result[i].join(" "));
}

File: BoSeonPark/최단경로/11404.py
import sys
import math
input = sys.stdin.readline

n = int(input())
m = int(input())

dp = [[math.inf]*(n+1) for _ in range(n+1)]

for _ in range(m):
    a, b, c = map(int, input().split())
    dp[a][b] = min(dp[a][b], c)

for j in range(1, n+1):
    for i in range(1, n+1):
        for k in range(1, n+1):
            if i == k:
                dp[i][k] = 0
            else:
                dp[i][k] = min(dp[i][j] + dp[j][k], dp[i][k])

for i in range(1, n+1):
    for j in range(1, n+1):
        if dp[i][j] == math.inf:
            print(0, end=' ')
        else:
            print(dp[i][j], end=' ')
    print('')